Examples/laser_acceleration/laser_acceleration_2d_PICMI.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+
+# 2D version of the LWFA example, provided for quick debugging
+# This should be the only line that needs to be changed for different codes
+# e.g. `from pywarpx import picmi`
+#      `from fbpic import picmi`
+#      `from warp import picmi`
+from plasmacode import picmi
+
+# Create alias fror constants
+cst = picmi.constants
+
+# Run parameters - can be in separate file
+# ========================================
+
+# Physics parameters
+# ------------------
+
+# --- laser
+import math
+laser_polarization   = math.pi/2 # Polarization angle (in rad)
+laser_a0             = 4.        # Normalized potential vector
+laser_wavelength     = 8e-07     # Wavelength of the laser (in meters)
+laser_waist          = 5e-06     # Waist of the laser (in meters)
+laser_duration       = 15e-15    # Duration of the laser (in seconds)
+laser_injection_loc  = 9.e-6     # Position of injection (in meters, along z)
+laser_focal_distance = 100.e-6   # Focal distance from the injection (in meters)
+laser_t_peak         = 30.e-15   # The time at which the laser reaches its peak
+                                 # at the antenna injection location (in seconds)
+# --- plasma
+plasma_density_expression = "1.e23*(1+tanh((y - 20.e-6)/10.e-6))/2."
+plasma_min     = [-20.e-6,  0.0e-6]
+plasma_max     = [ 20.e-6,  1.e-3]
+
+# --- electron bunch
```

Comment:
The intent of the Gaussian beam distribution is that it is describing the physics of the beam and is not dependent on the dimensionality of the underlying simulation. So, the input parameters should all have lengths of 3, and the number of physical particles is well defined.

In lower dimensional simulations, the ignored directions would have a unit length.